File: ShadowFrameworkLite20/src/shadow/pipeline/expression/SFExpressionIndex.java
package shadow.pipeline.expression;

import shadow.math.SFValuenf;
import shadow.pipeline.parameters.SFParameteri;

public class SFExpressionIndex extends SFExpressionElement {

	private int index;
	
	public SFExpressionIndex(int index) {
		super(""+index);
		this.index=index;
		setType(SFParameteri.GLOBAL_GENERIC);
	}

	@Override
	public void addSubExpression(SFExpressionElement element) throws ArrayIndexOutOfBoundsException {
		// Nothing to do
	}
	
	@Override
	public SFExpressionElement cloneAsIndexed(SFParameteri[] toBeIndexed) {
		return this;
	}

	@Override
	public void evaluateType() throws SFExpressionException {
		// Nothing to do..
	}

	@Override
	public SFValuenf evaluate(SFExpressionValuesList values) {
		return values.getValue(index);
	}

}

File: trunk/ShadowFrameworkJS/src/shadow/pipeline/expression/SFExpressionIndex.js



function SFExpressionIndex(index){
	this.element = ""+index;
	this.index=index;
	this.setType(SFParameteri_GLOBAL_GENERIC);
}

inherit(SFExpressionIndex,SFExpressionElement);


SFExpressionIndex.prototype["addSubExpression"]=function(element){
			// Nothing to do
		};

SFExpressionIndex.prototype["evaluateType"]=function(element){
			// Nothing to do
		};

SFExpressionIndex.prototype["cloneAsIndexed"]=function(toBeIndexed){
			return this;
		};


SFExpressionIndex.prototype["evaluate"]=function(values){
			return values.getValue(this.index);
		};

